specs/merge/beacon-chain.md

```diff
@@ -209,41 +197,38 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
     process_randao(state, block.body)
     process_eth1_data(state, block.body)
     process_operations(state, block.body)
-    # Pre-merge, skip execution payload processing
-    if is_execution_enabled(state, block):
-        process_execution_payload(state, block.body.execution_payload, EXECUTION_ENGINE)  # [New in Merge]
+    process_execution_payload(state, block.body)  # [New in Merge]
 ```
 
-#### Execution payload processing
+### Execution payload processing
 
-##### `process_execution_payload`
+#### `process_execution_payload`
 
 ```python
-def process_execution_payload(state: BeaconState,
-                              execution_payload: ExecutionPayload,
-                              execution_engine: ExecutionEngine) -> None:
-    """
-    Note: This function is designed to be able to be run in parallel with the other `process_block` sub-functions
-    """
-    if is_transition_completed(state):
-        assert execution_payload.parent_hash == state.latest_execution_payload_header.block_hash
-        assert execution_payload.number == state.latest_execution_payload_header.number + 1
-
-    assert execution_payload.timestamp == compute_time_at_slot(state, state.slot)
-
-    assert execution_engine.new_block(execution_payload)
-
+def process_execution_payload(state: BeaconState, body: BeaconBlockBody) -> None:
+    # Skip execution payload processing if execution is not enabled
+    if not is_execution_enabled(state, body):
```

Comment:
We previously pulled this block condition *out* because it isn't really part of "processing the execution payload" and majorly complicates testing

Reply:
> it isn't really part of "processing the execution payload"

The `if` statement is part of the other `process_` functions, e.g.:

<img width="599" alt="Screenshot 2021-06-08 at 15 56 31" src="https://user-images.githubusercontent.com/731743/121208609-193a2500-c872-11eb-8124-0aa853238e52.png">

<img width="538" alt="Screenshot 2021-06-08 at 15 57 15" src="https://user-images.githubusercontent.com/731743/121208758-33740300-c872-11eb-98b5-79c5b514faf2.png">

<img width="505" alt="Screenshot 2021-06-08 at 15 59 06" src="https://user-images.githubusercontent.com/731743/121209091-759d4480-c872-11eb-9ef5-1acffbcc479a.png">

Don't feel too strongly either way.